jubilant/_juju.py

```diff
@@ -23,7 +26,15 @@ def __str__(self):
 
 
 class WaitError(Exception):
-    """Raised when :meth:`Juju.wait`'s "error" callable returns False."""
+    """Raised when :meth:`Juju.wait`'s *error* callable returns False."""
+
+
+class SecretURI(str):
+    """A string subclass that represents a secret URI ("secret:...")."""
+
+
+type ConfigValue = bool | int | float | str | SecretURI
+"""The possible types a charm config value can be."""
```

Comment:
I couldn't for the life of my get Sphinx showing these how I wanted them, or to get ConfigValue to show up in the docs at all. I tried using [`autodoc_type_aliases`](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#confval-autodoc_type_aliases), but it was acting up. I think I'll wait for David's Sphinx/docs overhaul and then play with it a bit more.